src/bbtag/compilation/compileSignatures.ts

```diff
@@ -43,10 +46,12 @@ export function compileSignatures(signatures: readonly SubtagSignatureCallable[]
 }
 
 function createConditionalHandler(signature: SubtagSignatureCallable, resolver: ArgumentResolver): ConditionalSubtagHandler {
+    const name = signature.subtagName?.toLowerCase();
+
     return {
-        canHandle(subtag) {
-            return resolver.isExactMatch(subtag);
-        },
+        canHandle: name === undefined
+            ? subtag => resolver.isExactMatch(subtag)
+            : (subtag, subtagName) => subtagName.toLowerCase() === name && resolver.isExactMatch(subtag),
```

Comment:
what's the point in a ternary operation here? you could merge the conditions together and only have it return the boolean result from everything with some || and &&

Reply:
The idea is that the body of this function is a hot path, every subtag call will call whatever this function is. The name never changes and we dont care what the subtag name is if the signatures subtag name is undefined, so why check it every time?

Its a bit of a micro optimisation for sure, but we want this as fast as possible